src/main/java/uk/nhs/digital/nhsconnect/nhais/config/NhaisMongoClientConfiguration.java

```diff
@@ -0,0 +1,89 @@
+package uk.nhs.digital.nhsconnect.nhais.config;
+
+
+import com.google.common.base.Strings;
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.mongodb.config.AbstractMongoClientConfiguration;
+
+@Configuration
+@Slf4j
+public class NhaisMongoClientConfiguration extends AbstractMongoClientConfiguration {
+
+    @Value("${nhais.mongodb.database}")
+    private String database;
+
+    @Value("${nhais.mongodb.uri}")
+    private String uri;
+
+    @Value("${nhais.mongodb.host}")
+    private String host;
+
+    @Value("${nhais.mongodb.port}")
+    private String port;
+
+    @Value("${nhais.mongodb.username}")
+    private String username;
+
+    @Value("${nhais.mongodb.password}")
+    private String password;
+
+    @Value("${nhais.mongodb.options}")
+    private String options;
+
+    @Value("${nhais.mongodb.autoIndexCreation}")
+    private String autoIndexCreation;
+
+    @Override
+    public String getDatabaseName() {
+        return this.database;
+    }
+
+    @Override
+    protected void configureClientSettings(MongoClientSettings.Builder builder) {
```

Comment:
This method is not called when integration tests are run using the `docker -v ...` command from Jenkins pipeline. You can tell because the logs messages are not printed. When running the tests from intellij or `./gradlew check` you can see that `configureClientSettings` is called and the overridden configuration is loaded.

Reply:
This has somehow resolved itself